code/game/objects/items/weapons/switchtool.dm

```diff
@@ -207,3 +216,178 @@
+			stored_modules["/obj/item/weapon/wirecutters:Wirecutters"] = new /obj/item/weapon/wirecutters(src)
+			stored_modules["/obj/item/weapon/crowbar:Crowbar"] = new /obj/item/weapon/crowbar(src)
+			stored_modules["/obj/item/device/multitool:Multitool"] = new /obj/item/device/multitool(src)
+			stored_modules["/obj/item/weapon/weldingtool/experimental:Weldingtool"] = new /obj/item/weapon/weldingtool/experimental(src)
+			to_chat(user, "The Holo switchtool has engineering designs now!")
+			return
+	if(istype(D, /obj/item/weapon/disk/tech_disk))
+		var/obj/item/weapon/disk/tech_disk/T = D
+		var/datum/tech/disk_tech = T.stored
+		if(istype(disk_tech, /datum/tech/combat) && disk_tech.level >= 5)
+			stored_modules["/obj/item/weapon/shield/energy:Shield"] = new /obj/item/weapon/shield/energy(src)
+			to_chat(user, "The Holo switchtool has a defensive design now!")
+			return
+	if(istype(D, /obj/item/weapon/disk/tech_disk))
+		var/obj/item/weapon/disk/tech_disk/T = D
+		var/datum/tech/disk_tech = T.stored
+		if(istype(disk_tech, /datum/tech/syndicate) && disk_tech.level >= 3)
+			stored_modules["/obj/item/weapon/melee/energy/sword/activated:Sword"] = new /obj/item/weapon/melee/energy/sword/activated(src)
+			to_chat(user, "The Holo switchtool has an offensive design now!")
+			return
+	if(istype(D, /obj/item/weapon/disk/tech_disk)) //easter egg sword based on the LoZ BoTW guardian++ sword, this is bus levels of research only. Didn't want to let the sprite go to waste.
+		var/obj/item/weapon/disk/tech_disk/T = D
+		var/datum/tech/disk_tech = T.stored
+		if(istype(disk_tech, /datum/tech/nanotrasen) && disk_tech.level >= 5)
+			stored_modules["/obj/item/weapon/melee/energy/hfmachete/activated:Sharper sword"] = new /obj/item/weapon/melee/energy/hfmachete/activated(src)
+			to_chat(user, "The Holo switchtool has a secret offensive design now!")
+			return
+	/**
+	//Joke module about power[clean/creep]
+	if(istype(D, /obj/item/weapon/disk/tech_disk))
+		var/obj/item/weapon/disk/tech_disk/T = D
+		var/datum/tech/disk_tech = T.stored
+		if(istype(disk_tech, /datum/tech/powerstorage) && disk_tech.level >= 4)
+			stored_modules["/obj/item/weapon/soap:Soap"] = new /obj/item/weapon/circular_saw(src)
+			to_chat(user, "The Holo switchtool has power clean designs now!")
+			return
+	**/
+//Since you can't turn off the welder inside the tool, I'm using the unused welder that very slowly regens fuel, looks like 1u per 5 byond seconds, thanks byond.
+//It can be refulled manually, but since it starts active you will blow up welder tanks if deployed and then put to a tank.
+	if(istype(deployed, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/weldingtool = deployed
+		weldingtool.welding = 1
+		weldingtool.status = 1
+		weldingtool.max_fuel = 50
+		weldingtool.start_fueled = 1
+
+//for the inhand sprite changes
+//Big shitty wall of else if lazy code
+
+obj/item/weapon/switchtool/holo/update_icon()
+	if(istype(deployed, /obj/item/device/flashlight))
```

Comment:
This giant wall of `if()` statements could be cleaned up by instead making an associative list of item type keys and `item_state` values, so you could just do `item_state = inhandlist[deployed.type]`.